src/Preflight/Preflight.php

```diff
@@ -202,6 +223,11 @@ public function preflight($argv)
         $this->preflightArgs = $this->preflightArgs($argv);
         $this->prepareConfig($this->environment);
 
+        // Setup our early logger.
+        $logger = new PreflightLog();
+        $logger->setDebug($this->preflightArgs->get(PreflightArgs::DEBUG));
+        $this->setLogger($logger);
```

Comment:
Usually DI happens from outside the class. Not sure if we want to add another object instantiation to drush_main (or whatever we're calling it) tho

Reply:
Not sure I follow. This is Preflight and PreflightLog. Two different classes.

Reply:
Yes, `--debug` is defined as a global option; however, if you run `drush status --debug`, it no longer turns on debug mode for most of Drush. The value is only set for preflight.